### DISCUSSION TITLE
Given an OffsetDateTime, how to find the beginning of the current month and beginning of the next month?

### DISCUSSION THREAD

**Discussion by discosultan** [time-rs/time#376] [Question] [answered]
For example, given an `OffsetDateTime` ("2020-12-26T21:02:00Z"), how to:

1. Get an `OffsetDateTime` that represents the beginning of the current month ("2020-12-01T00:00:00Z")?
2. Get an `OffsetDateTime` that represents the beginning of the next month ("2021-01-01T00:00:00Z")?

With Chrono crate, I was able to achieve this using the `DateTime::date`, `DateTime::from_utc`, `Date::with_day`, `Date::with_time`, `Date::from_ymd` APIs. Haven't figured it out for this crate yet.

**Comment by jhpratt** [comment, marked as answer]
On the assumption that chrono's `DateTime` is the same as time's `OffsetDateTime`:

You can get the year, month, and day by calling `OffsetDateTime::to_calendar_date`. To get the beginning of the month, call ~~`OffsetDateTime::from_calendar_date`~~ `Date::from_calendar_date` with the day set to one. For the next month, you can use `Month::next` and set the day to one. For that, you'll have to manually adjust the year (should be as easy as checking if the month is December).

**Reply by discosultan** [reply]
Thanks, that did the trick! Just a note that `OffsetDateTime::from_calendar_date` should be `Date::from_calendar_date` instead. I then constructed an `OffsetDateTime` by using `Date::with_hms` and `PrimitiveDateTime::assume_offset`.

**Reply by jhpratt** [reply]
Ah, right. I typed it off-hand without thinking about the specifics.
